piecrust/src/vm.rs

```diff
@@ -29,7 +30,7 @@ const SESSION_COMMITS_FILENAME: &str = "commits";
 struct VMInner {
     host_queries: HostQueries,
     base_memory_path: PathBuf,
-    session_commits: SessionCommits,
+    session_commits: RefCell<SessionCommits>,
```

Comment:
Instead of using VMInner and a RefCell, just make the borrow &mut in  Session, as requested in the issue.